bridge-token-factory/src/lib.rs

```diff
@@ -159,13 +222,36 @@ pub trait ExtBridgeToken {
         icon: Option<String>,
     );
 
+    fn ft_transfer(&mut self, receiver_id: AccountId, amount: U128, memo: Option<String>);
+
     fn set_paused(&mut self, paused: bool);
+
+    fn upgrade_and_migrate(&mut self, code: &[u8]);
 }
 
 pub fn assert_self() {
     assert_eq!(env::predecessor_account_id(), env::current_account_id());
 }
 
+/*
+Checks whether the `withdrawer` is an aurora-chain account
+returns true iff `withdrawer` is an aurora account.
+ */
+pub fn is_aurora_engine_account(withdrawer: AccountId) -> bool {
```

Comment:
It is better to store the known engine accounts like in the [aurora-eth-connector](https://github.com/aurora-is-near/aurora-eth-connector/blob/master/eth-connector/src/lib.rs#L57)
known_engine_accounts

Reply:
And probably this can be combined with `withdraw_fee_percentage_per_silo`, so one map should be enough

Reply:
So we will only take fee as per silo, only for engine accounts which are configured with default or token-wise fee? If not set, we take fee whatever is set for aurora transfers.